docs/en/tutorials/getting-started.rst

```diff
@@ -233,33 +233,154 @@ entity definition:
+
+.. note::
+
+    This method, although very common, is **not** the best approach if
+    you want a perfect domain design. It can lead to unexpected behavior
+    when dealing with different steps of logic in between different
+    calls to ``EntityManager::flush()``, for example when you change
+    the state of an object via event listeners or lifecycle callbacks.
+
+**Mutators and DTOs**
+
+We recommend using a rich model design and rely on more complex
+mutators, ideally based on DTOs.
+This way, you should **not** use getters nor setters, and instead,
+implement methods that represent the **behavior** of your domain.
+
+For example, when having a ``User`` entity, we could foresee
+the following kind of optimization.
+
+Before, an anemic model with getters and setters:
+
+.. configuration-block::
+
+    .. code-block:: php
+
+        <?php
+        class User {
+            private $username;
+            private $passwordHash;
+            private $bans;
+
+            public function getUsername() :string {
+                return $this->username;
+            }
+
+            public function setUsername(string $username) {
+                $this->username = $username;
+            }
+
+            public function getPasswordHash() : string {
+                return $this->passwordHash;
+            }
+
+            public function setPasswordHash(string $passwordHash) {
+                $this->passwordHash = $passwordHash;
+            }
+
+            public function addBan(Ban $ban) {
+                $this->bans[] = $ban;
+            }
         }
 
-        public function getName()
+After, a rich model with proper accessors and mutators:
+
+.. configuration-block::
+
+    .. code-block:: php
+        <?php
+        class User
         {
-            return $this->name;
+            private $banned;
+            private $username;
+            private $passwordHash;
+            private $bans;
+
+            public function toNickname() : string
+            {
+                return $this->username;
+            }
+
+            public function authenticate(string $password, callable $checkHash) : bool
+            {
+                return $checkHash($password, $this->passwordHash) && ! $this->hasActiveBans();
+            }
+
+            public function changePass(string $password, callable $hash)
+            {
+                $this->passwordHash = $hash($password);
+            }
+
+            public function ban()
+            {
+                $this->bans[] = new Ban($this);
+            }
         }
 
-        public function setName($name)
+Additionnally, our entities should never see their state change
+partially without validation. For example, creating a ``new Product()``
+object without any data makes it an **invalid object**.
+Entities should represent **behavior**, not **data**, therefore
+they should be valid even after a ``__construct()``.
+
+To help creating such objects, we can rely on ``DTO``s, and/or make
```

Comment:
>  make our entities Value Objects.
Can we rephrase it? From DDD point of view VOs and entities are different concepts.